components/tikv_util/src/store/mod.rs

```diff
@@ -9,6 +9,78 @@ pub use self::{
     query_stats::{is_read_query, QueryStats},
     region::{
         check_key_in_region, check_key_in_region_exclusive, check_key_in_region_inclusive,
-        region_on_same_stores,
+        region_on_same_stores, region_on_stores,
     },
 };
+
+#[cfg(test)]
+mod tests {
+    use kvproto::metapb::Region;
+
+    use super::*;
+
+    #[test]
+    fn test_on_same_store() {
+        let cases = vec![
+            (vec![2, 3, 4], vec![], vec![1, 2, 3], vec![], false),
+            (vec![2, 3, 1], vec![], vec![1, 2, 3], vec![], true),
+            (vec![2, 3, 4], vec![], vec![1, 2], vec![], false),
+            (vec![1, 2, 3], vec![], vec![1, 2, 3], vec![], true),
+            (vec![1, 3], vec![2, 4], vec![1, 2], vec![3, 4], false),
+            (vec![1, 3], vec![2, 4], vec![1, 3], vec![], false),
+            (vec![1, 3], vec![2, 4], vec![], vec![2, 4], false),
+            (vec![1, 3], vec![2, 4], vec![3, 1], vec![4, 2], true),
+        ];
+
+        for (s1, s2, s3, s4, exp) in cases {
+            let mut r1 = Region::default();
+            for (store_id, peer_id) in s1.into_iter().zip(0..) {
+                r1.mut_peers().push(new_peer(store_id, peer_id));
+            }
+            for (store_id, peer_id) in s2.into_iter().zip(0..) {
+                r1.mut_peers().push(new_learner_peer(store_id, peer_id));
+            }
+
+            let mut r2 = Region::default();
+            for (store_id, peer_id) in s3.into_iter().zip(10..) {
+                r2.mut_peers().push(new_peer(store_id, peer_id));
+            }
+            for (store_id, peer_id) in s4.into_iter().zip(10..) {
+                r2.mut_peers().push(new_learner_peer(store_id, peer_id));
+            }
+            let res = region_on_same_stores(&r1, &r2);
+            assert_eq!(res, exp, "{:?} vs {:?}", r1, r2);
+        }
+    }
+
+    #[test]
+    fn test_check_region_on_store() {
```

Comment:
Better to put this test to `region.rs` and not move the above test.

Reply:
The reason why I check and modify the uts from `regions.rs` into `mod.rs`is that, the original ut in `peer.rs` includes function callings from `region.rs`.
https://github.com/tikv/tikv/blob/92aaf95e0dc83f855cf6d9bfd046c2ea0dfc9f3d/components/tikv_util/src/store/peer.rs#L96

IMO, it's more approapriate to assemble these special uts into `mod.rs`.